spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/PropertiesKafkaConnectionDetails.java

```diff
@@ -64,7 +65,7 @@ private List<Node> asNodes(List<String> bootstrapServers) {
 	}
 
 	private Node asNode(String bootstrapNode) {
-		int separatorIndex = bootstrapNode.indexOf(':');
+		int separatorIndex = bootstrapNode.lastIndexOf(':');
 		if (separatorIndex == -1) {
 			return new Node(bootstrapNode, this.DEFAULT_PORT);
 		}
```

Comment:
For a bootstrapNode with a value of "PLAINTEXT://localhost:9092" this implementation will return a Node with a host of "PLAINTEXT://localhost" and port of "9092". I am not sure that the host part is ok since I guess it should be just "localhost".

Reply:
I think it should include protocol part to restore server node from `Node` object, maybe the `host` should rename to a more meaningful name.